src/main/java/net/coderbot/iris/postprocess/CenterDepthSampler.java

```diff
@@ -1,62 +1,107 @@
+		RenderSystem.bindTexture(texture);
+		setupColorTexture(format);
+		RenderSystem.bindTexture(altTexture);
+		setupColorTexture(format);
+		RenderSystem.bindTexture(0);
 
-		// NB: This will always be one frame behind compared to the current frame.
-		// That's probably for the best, since it can help avoid some pipeline stalls.
-		// We're still going to get stalls, though.
-		centerDepthSmooth = new SmoothedFloat(1.0f, 1.0f, this::sampleCenterDepth, fakeNotifier);
-
-		// Prior to OpenGL 4.1, all framebuffers must have at least 1 color target.
-		depthBufferHolder = renderTargets.createFramebufferWritingToMain(new int[] {0});
-		this.renderTargets = renderTargets;
-	}
-
-	public void updateSample() {
-		fakeNotifier.onNewFrame();
+		this.framebuffer.addColorAttachment(0, texture);
+		ProgramBuilder builder = ProgramBuilder.begin("centerDepthSmooth", "#version 120\n" +
+			" void main() { gl_Position = ftransform(); }", null, "#version 120\n" +
+			" uniform sampler2D depth; \n" +
+			" uniform sampler2D altDepth; \n" +
+			" uniform float lastFrameTime; \n" +
+			" uniform float decay; \n" +
+			" void main() { float currentDepth = texture2D(depth, vec2(0.5)).r; float decay2 = 1.0 - exp(-decay * lastFrameTime); gl_FragColor = vec4(mix(texture2D(altDepth, vec2(0.5)).r, currentDepth, decay2), 0, 0, 0); }", ImmutableSet.of());
+		builder.addDynamicSampler(() -> Minecraft.getInstance().getMainRenderTarget().getDepthTextureId(), "depth");
+		builder.addDynamicSampler(() -> altTexture, "altDepth");
+		builder.uniform1f(UniformUpdateFrequency.PER_FRAME, "lastFrameTime", SystemTimeUniforms.TIMER::getLastFrameTime);
+		builder.uniform1f(UniformUpdateFrequency.ONCE, "decay", () -> (1.0f / ((halfLife * 0.1) / LN2)));
+		this.program = builder.build();
 	}
 
-	private float sampleCenterDepth() {
+	public void sampleCenterDepth() {
 		if (hasFirstSample && (!everRetrieved)) {
 			// If the shaderpack isn't reading center depth values, don't bother sampling it
 			// This improves performance with most shaderpacks
-			return 0.0f;
+			return;
 		}
 
 		hasFirstSample = true;
 
-		this.depthBufferHolder.bind();
+		this.framebuffer.bind();
+		this.program.use();
+
+		RenderSystem.viewport(0, 0, 3, 3);
+
+		FullScreenQuadRenderer.INSTANCE.render();
+
+		GlStateManager._glUseProgram(0);
+
+		GL11C.glDisable(GL11C.GL_SCISSOR_TEST);
 
-		float[] depthValue = new float[1];
-		// Read a single pixel from the depth buffer
-		// TODO: glReadPixels forces a full pipeline stall / flush, and probably isn't too great for performance
-		IrisRenderSystem.readPixels(
-			renderTargets.getCurrentWidth() / 2, renderTargets.getCurrentHeight() / 2, 1, 1,
-			GL11C.GL_DEPTH_COMPONENT, GL11C.GL_FLOAT, depthValue
-		);
+		this.framebuffer.bind();
 
-		return depthValue[0];
+		GlStateManager._bindTexture(altTexture);
+		DepthCopyStrategy.fastest(false).copy(this.framebuffer, texture, null, altTexture, 3, 3);
```

Comment:
The API contract of DepthCopyStrategy says it can only copy depth... This will work because the parameter is false, but this deserves a sort of TODO comment addressing this - it should state that it still works even despite the issues.